### PR TITLE
openssh: add version 9.4p1 and fix compile issue with new zlib-1.3

### PR DESCRIPTION
**Pull request by bernhardkaindl**
Update to 9.4p1 by @snehring (also independently tested by me).
I improved the fix for the zlib versions

I am the maintainer of this package and added:
- add missing `openssl` dependency version `limit` to older `openssh` versions
- add missing `-fcommon` for new compilers building old `openssh` versions

Fully tested by me with all relevant versions.

### CODE REVIEW

**Comment by haampie** [comment]
Fine with this, but can you make CFLAGS extensible? For example:

```python
def setup_build_environment(self, env):
    if self.spec.satisfies("@:7 %gcc@10:") or self.spec.satisfies("@:7 %clang@11:"):
        env.append_flags("CFLAGS", "-fcommon")
```

File: var/spack/repos/builtin/packages/openssh/package.py [hunk 1]
```diff
@@ -54,6 +55,8 @@ class Openssh(AutotoolsPackage):
 
     depends_on("krb5+shared", when="+gssapi")
     depends_on("openssl@:1.0", when="@:7.7p1")
+    depends_on("openssl@:1.1", when="@:7.9p1")
+    depends_on("openssl@:3.0", when="@:8.7p1")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/openssh/package.py:60]: Probably you mean `@:3` here? OpenSSL 3 follows semver.

Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/openssh/package.py:60]: **Nope, semver has nothing to with that:**

OpenSSH-8.7 works with OpenSSL 3.0 but don't support 3.1.

**Comment by bernhardkaindl** [comment]
> Fine with this, but can you make CFLAGS extensible? For example:
> 
> ```python
> def setup_build_environment(self, env):
>     if self.spec.satisfies("@:7 %gcc@10:") or self.spec.satisfies("@:7 %clang@11:"):
>         env.append_flags("CFLAGS", "-fcommon")
> ```

Done! Updated for your approval!

File: var/spack/repos/builtin/packages/openssh/package.py (hunk 1 above)
Comment (haampie) [reply var/spack/repos/builtin/packages/openssh/package.py:60]: Then I guess it's a faulty configure script, like with zlib, but let's leave it :)

**Review by haampie — APPROVED** [review]

**Pull request merged**